test/behavior/packed_struct_equality.zig

```diff
@@ -0,0 +1,13 @@
+const std = @import("std");
+const expect = std.testing.expect;
+
+test "packed struct equality" {
+    const S = packed struct {
+        a: u4,
+        b: u4,
+    };
+    const x: S = .{ .a = 1, .b = 2 };
+    const y: S = .{ .b = 2, .a = 1 };
+    try expect(x == y);
+    try expect(!(x != y));
```

Comment:
please also test runtime

Reply:
yep, good point. forgot.

Reply:
done, and split into two clearer commits.